do.py

```diff
@@ -150,22 +153,20 @@ def test():
     go_coverage_threshold = 35
     # TODO: not able to run the test from main directory
     os.chdir("pkg")
-    try:
-        run(["go test ./... -v -coverprofile coverage.txt | tee coverage.out"])
-    finally:
-        os.chdir("..")
-
-    with open("pkg/coverage.out") as fp:
-        out = fp.read()
-        result = re.findall(r"coverage:.*\s(\d+)", out)[0]
-        if int(result) < go_coverage_threshold:
-            raise Exception(
-                "Go tests achieved {1}% which is less than Coverage thresold {0}%,".format(
-                    go_coverage_threshold, result))
-        else:
-            print(
-                "Go tests achieved {1}% ,Coverage thresold {0}%".format(
-                    go_coverage_threshold, result))
+    ret = subprocess.run("go test ./... -v -coverprofile coverage.txt".split(), capture_output=True)
+    print(ret.stdout.decode("utf-8"))
+    if b'FAIL' in ret.stdout:
+        raise Exception("Go Tests Failed")
```

Comment:
This will always raise error even if there's only one TC that failed. We actually want to fail only when coverage report is below threshold.

Reply:
yes, Ideally if one TC fails the CICD shall fail right. in the last CICD you noticed even though the testcases failed the coverage was beyond the threshold.

Reply:
then we need to fix the threshold value. otherwise there's no point in measuring coverage (because we'll fail in advance)

Reply:
I think TC validation shall get high priority than coverage, if testcases are failing there is no point in checking the coverage

Reply:
I mean we can move exception checking part to end (i.e. after coverage check)